### ISSUE TITLE
bug: yy (and other actions) crash nvim with error code 132 (WSL)

### ISSUE THREAD

**Issue by hannenz** [LazyVim/LazyVim#4104]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.11.0-dev-359+g5b778a64e-dirty

### Operating system/version

WSL Ubuntu 22.04 on Windows 10

### Describe the bug

pressing `yy` (Yank whole line) exits nvim with error code 132. I observed the same behavior with `dG` some times but with `yy` it is 100% reproducable. 

I tried wiping all existing nvim related stuff and did a fresh install of LazyVim and also tried the docker installation. The issue does NOT occur on the very first run but as soon as I enter nvim for the second time. 

The issue does NOT occur on a fresh LazyVim installation on my Linux machine but only in WSL on Windows 10, regardless of the terminal I use (Windows Terminal or Alacritty). The behavior occurs regardless if editing an existing file or a new buffer, having zero or more text in it, if the line is empty or not.

It does NOT occur with the minimal init.lua (`nvim -u repro.lua`)

### Steps To Reproduce

1. git clone https://github.com/LazyVim/starter ~/.config/nvim
2. Enter `nvim`
3. type `n` for a new file / buffer
4. Enter normal mode with `ESC`
5. Type `yy`
6. Editor exits to shell with exit code 132

### Expected Behavior

Editor does not exit with exit code 132

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- add any other plugins here
  },
})
```

**Comment by folke** [comment]
Before I look into this, can you please update which-key and try again?
I made some changes in the last hour or so that might fix this.

**Comment by folke** [comment]
If the latest changes don't work, then please shiw the output of `:checkhealth vim.provider`.
Your clipboard config is probably causing this.

**Comment by hannenz** [comment]
I updated `which-key` and the problem persists. Fun fact: While copying the output of `checkhealth vim.provider` with `"+yG` the bug occured as well, but the content was in the clipboard:

```
==============================================================================
vim.provider: require("vim.provider.health").check()

Clipboard (optional) ~
- OK Clipboard tool found: xclip

Node.js provider (optional) ~
- Node.js: v19.9.0
  
- WARNING Missing "neovim" npm (or yarn, pnpm) package.
  - ADVICE:
    - Run in shell: npm install -g neovim
    - Run in shell (if you use yarn): yarn global add neovim
    - Run in shell (if you use pnpm): pnpm install -g neovim
    - You may disable this provider (and warning) by adding `let g:loaded_node_provider = 0` to your init.vim

Perl provider (optional) ~
- WARNING "Neovim::Ext" cpan module is not installed
  - ADVICE:
    - See :help |provider-perl| for more information.
    - You may disable this provider (and warning) by adding `let g:loaded_perl_provider = 0` to your init.vim
- WARNING No usable perl executable found

Python 3 provider (optional) ~
- `g:python3_host_prog` is not set. Searching for python3 in the environment.
- Executable: /usr/bin/python3
- Python version: 3.10.12
- pynvim version: 0.5.0
- OK Latest pynvim is installed.

Python virtualenv ~
- OK no $VIRTUAL_ENV

Ruby provider (optional) ~
- Ruby: ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]
- WARNING `neovim-ruby-host` not found.
  - ADVICE:
    - Run `gem install neovim` to ensure the neovim RubyGem is installed.
    - Run `gem environment` to ensure the gem bin directory is in $PATH.
    - If you are using rvm/rbenv/chruby, try "rehashing".
    - See :help |g:ruby_host_prog| for non-standard gem installations.
    - You may disable this provider (and warning) by adding `let g:loaded_ruby_provider = 0` to your init.vim

```

**Comment by folke** [comment]
Does the issue happen when doing `:norm! yy`?

**Comment by folke** [comment]
And if you disable which-key?

**Comment by dpetka2001** [comment]
I found this [issue at Neovim](https://github.com/neovim/neovim/issues/20396) which also had a link to [another issue at microsoft terminal](https://github.com/microsoft/terminal/issues/11945) and found some users suggesting that `xsel` behaved better in WSL. Could you try that instead? Also there's a recommendation in [Neovim site for clipboard provider in WSL](https://neovim.io/doc/user/provider.html#clipboard-wsl) if you'd like to test out.

**Comment by folke** [comment]
Can you try setting `vim.o.clipboard = ""` in your `lua/config/options.lua`?
It things work then, then it's for sure an issue with your clipboard.

**Comment by hannenz** [comment]
I tried your recommendations (disabled `which-key`, `:norm! yy`, `vim.o.clipboard = ""` and also the `clipboard-wsl` but unfortunately still no luck.
The issues that you have linked to do not seem related to me, they talk about freezing nvim for a period of time, not crashing. Firthermore they are not related to `yy` but more about pasting from windows system clipbord into nvim.

**Comment by folke** [comment]
Can you be more specific than no luck with each of those things?

Or do you mean it still crashed for every thing you tried?

**Comment by dpetka2001** [comment]
Yes the issues didn't say explicitly that they got an E132 error, but they talked about Neovim hanging and they even happened at when pressing `yy` to yank a line (they were not explicit to pasting only, at least for some users), so I thought there was a chance they might be related. Did you also try to install `xsel` in WSL and uninstall `xclip` to see if that makes any difference? Also another user [commented that win32yank did the trick for him](https://github.com/microsoft/terminal/issues/11945#issuecomment-1441112078). These are various suggestions to just try and see if maybe one of them ends up working for you.

**Comment by folke** [comment]
Closing since this is not really a LazyVim issue.

**Comment by kev-chien** [comment]
I was running `NVIM v0.11.0-dev-657+gbb4b6b427` (from [neovim-releases](https://github.com/neovim/neovim-releases)), I also ran into this issue when using `LazyVim`, with exit code 132 when I use `yy`.
I'm using Rocky Linux 8.
When I started nvim with no config (no LazyVim), I was able to `yy` fine.
It happened regardless of what clipboard configuration I had: `tmux`, or none.
So, it appears it may be a combination of nvim v0.11.0, and LazyVim.

When I downgraded to `NVIM v0.10.1`, I found `yy` was able to work. That works for my needs, so sticking with that.
